kikuchipy/indexing/merge_crystalmaps.py

```diff
@@ -0,0 +1,132 @@
+    SimilarityMetric,
+    SIMILARITY_METRICS,
+)
+
+
+def merge_crystalmaps(
+    xmaps: List[CrystalMap],
+    mean_n_largest: int = 1,
+    metric: Union[str, SimilarityMetric] = None,
+):
+    """Merge a list of `CrystalMap`s into a single map. Used on results from DictionaryIndexing.
+
+    The given `CrystalMap`s must have "template_indices" and "metric_results" in prop
+    as produced by :class:`~kikuchipy.indexing.StaticDictionaryIndexing`.
+    Both props are merge sorted into the returned map.
+
+    Parameters
+    ----------
+    xmaps : List[CrystalMap]
+        List of `CrystalMap`s with "template_indices" and "metric_results" in prop
+    mean_n_largest : int, optional
+        Number of top metric results to take the mean of,
+        before comparing and put in the property `merged_top_metric_result`, by default 1
+    metric : Union[str, SimilarityMetric], optional
+        Similarity metric, by default None
+
+    Returns
+    -------
+    merged_xmap : CrystalMap
+        A CrystalMap with added prop "merged_top_metric_result"
+
+    Notes
+    -----
+    `mean_n_largest` can be given with a negative sign if metric is not given
+    in order to choose the lowest valued metric results.
+
+    Raises
+    ------
+    ValueError
+        If `xmaps` contains CrystalMaps with equal phases.
+    """
+    if metric is None:
+        sign = copysign(1, mean_n_largest)
+        mean_n_largest = abs(mean_n_largest)
+    else:
+        sign = SIMILARITY_METRICS.get(metric, metric).sign
+
+    top_results_across_xmaps = np.array(
+        [
+            np.mean(xmap.metric_results[:, :mean_n_largest], axis=1)
+            for xmap in xmaps
+        ]
+    )
+
+    phase_id = np.argmax(sign * top_results_across_xmaps, axis=0)
+    merged_top_metric_result = np.choose(phase_id, top_results_across_xmaps)
+
+    metric_results = np.concatenate(
+        [xmap.metric_results for xmap in xmaps], axis=1
+    )
+
+    metric_result_sorted_indicies = np.argsort(
+        sign * -metric_results, kind="mergesort", axis=1
+    )
+
+    template_indices = np.concatenate(
+        [xmap.template_indices for xmap in xmaps],
+        axis=1,
+    )
+    template_indices = np.take_along_axis(
+        template_indices, metric_result_sorted_indicies, axis=1
+    )
+    metric_results = np.take_along_axis(
+        metric_results, metric_result_sorted_indicies, axis=1
+    )
+
+    prop = {
+        "merged_top_metric_result": merged_top_metric_result,
+        "template_indices": template_indices,
+        "metric_results": metric_results,
+    }
+
+    # The rotations can be examined more carefully, takes now all keep_n only for the top 1 phase
+    # Maybe not ideal
+    rotations_across_xmaps = np.array([xmap.rotations.data for xmap in xmaps])
+    rotations = np.array(
+        [rotations_across_xmaps[id][i] for i, id in enumerate(phase_id)]
+    )
+    rotations = Rotation(rotations)
+
+    # OBS! Can raise ValueError if phase already in phase_lists
```

Comment:
Now warns instead of raising ValueError.

Reply:
Hm, I'm reconsidering the choice to just throw a warning and not raise an error... The question is: should `merge_crystalmaps()` work if some of the phases in two or more different maps have the same phase?

Reply:
As a deep future take, I would say yes, they should merge. One could imagine situation where other things were changed between maps (processing/simulation choice etc). Only a note though, it can be changed when we get there obviously.

Reply:
I agree with you both that they should merge. It now works so that a phase with a duplicate phase name gets a new name by adding the index of its map's position in the xmaps array, e.g.: xmaps = [xmap_austenite, xmap_austenite] yields a PhaseList in the merged xmap with austenite and austenite2 as phases.